pkg/api/applications/v2/http.go

```diff
@@ -39,6 +39,23 @@ type httpAPI struct {
 
 var _ API = &httpAPI{}
 
+func (h *httpAPI) CheckEndpoint(ctx context.Context) (api.Metadata, error) {
+	req, err := http.NewRequest(http.MethodHead, h.client.URL(endpointApplications).String(), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, body, err := h.client.Do(ctx, req)
+	switch {
+	case api.IsUnauthorized(err):
+		return nil, err
+	case err != nil:
+		return nil, api.NewUnexpectedError(resp, body)
+	default:
+		return api.Metadata(resp.Header), nil
+	}
```

Comment:
I had to double take on this a couple of times because it doesn't fit the mold of the other implementations (where the switch is on the `resp.StatusCode`). I think it also has an issue in that we normally return non-`nil` errors from `Do` directly because if the `err` is not-`nil` it is probably also the case that `resp` and `body` are not actually usable.

In general, I would have expected:

```
resp, body, err := h.client.Do(ctx, req)
if err != nil {
  return nil, err
}

switch resp.StatusCode {

// Enumerate status codes from OpenAPI doc here, including the "success" case(s)

default:
  return nil, api.NewUnexpectedError(resp, body)
}
```

Since we don't necessarily have this in the OpenAPI spec, it ends up kind of like the old `OPTIONS` case where the "success" case may actually be a non-2xx HTTP status.